packages/core/lib/debug/cli.js

```diff
@@ -204,6 +205,42 @@ class CLIDebugger {
 
     return contracts;
   }
+
+  /**
+   * This function is responsible for opening the debugger in vscode.
+   */
+  async openVSCodeDebug() {
+    // Sets the URL
+    const url = new URL("/debug", "vscode://trufflesuite-csi.truffle-vscode");
+
+    // Sets the query parameters
+    url.searchParams.set("txHash", this.txHash);
+    url.searchParams.set("workingDirectory", this.config.working_directory);
+    url.searchParams.set("providerUrl", this.config.provider.host);
```

Comment:
Note that this is brittle, since we can't always rely on being able to get a URL from a provider